src/coreclr/src/debug/ee/arm64/arm64walker.cpp

```diff
@@ -187,81 +184,55 @@ BYTE*  NativeWalker::SetupOrSimulateInstructionForPatchSkip(T_CONTEXT * context,
-                    SimdRegContents.Low = RegContents;
-                    SimdRegContents.High = 0;
+                else // LDR Wt [imm]
+                    SetReg(context, RegNum, SimdRegContents.Low);
+
+                break;
+            case 1: //load 8 bytes
+                SimdRegContents.Low = GetMem(ip, 8, false);
+                SimdRegContents.High = 0;
+                if (isSimd) //LDR Dt [imm]
                     SetSimdReg(context, RegNum, SimdRegContents);
-                    break;
-                case 2: //SIMD 16 byte data
+                else // LDR Xt [imm]
+                    SetReg(context, RegNum, SimdRegContents.Low);
+                break;
+            case 2: //SIMD 16 byte data
+                if (isSimd) //LDR Qt [imm]
+                {
                     SimdRegContents = GetSimdMem(ip);
                     SetSimdReg(context, RegNum, SimdRegContents);
-                    break;
-                default:
-                    LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Simulate Unknown opcode: %x [LDR(litera,SIMD &FP)]  \n", opcode));
-                    _ASSERTE(!("Arm64Walker::Simulated Unknown opcode"));
-
                 }
-            }
-            else
-            {
-                short opc = (opcode >> 30);
-                switch (opc)
+                else //LDR St [imm] (signed)
                 {
-                case 0: //4byte data into Wt
-                    LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Simulate opcode: %x to LDR W%d %p\n", opcode, RegNum, offset));
-                    RegContents = 0xFFFFFFFF & RegContents;  //zero the upper 32bits
-                    SetReg(context, RegNum, RegContents);
-                    break;
-
-                case 1: //8byte data into Xt
-                    LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Simulate opcode: %x to LDR X%d %p\n", opcode, RegNum, offset));
-                    SetReg(context, RegNum, RegContents);
-                    break;
-
-                case 2: //LDRSW
-                    LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Simulate opcode: %x to LDRSW X%d %p\n", opcode, RegNum, offset));
-                    RegContents = 0xFFFFFFFF & RegContents;
-
-                    if (RegContents & 0x80000000) //Sign extend the Word
-                    {
-                        RegContents = 0xFFFFFFFF00000000 | RegContents;
-                    }
-                    SetReg(context, RegNum, RegContents);
-                    break;
-                case 3:
-                    LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Simulate opcode: %x  as PRFM ,but do nothing \n", opcode));
-
-                    break;
-                default:
-                    LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Simulate Unknown opcode: %x [LDR(literal)]  \n", opcode));
-                    _ASSERTE(!("Arm64Walker::Simulated Unknown opcode"));
-
+                    SimdRegContents.Low = GetMem(ip, 4, true);
```

Comment:
This needs to be sign extended.  Lines 241-244 above.

Reply:
That's what the true is for. Maybe I should add a comment.